File: src/giada/swing/MnR/DefaultHTMLImageProducer.java
package giada.swing.MnR;

import static def.dom.Globals.document;
import simulation.dom.$Image;

/**
 * The default implementation of the HTMLImageProducer
 *
 * @author gianpiero.diblasi
 * @param <T> The value type of the AbstractSliderModelAndRenderer
 */
public class DefaultHTMLImageProducer<T> implements HTMLImageProducer<T> {

  private final T value;
  private final String src;

  public DefaultHTMLImageProducer(T value, String src) {
    super();
    this.value = value;
    this.src = src;
  }

  @Override
  public $Image produce() {
    $Image img = ($Image) document.createElement("img");
    img.src = this.src;
    return img;
  }

  @Override
  public T getValue() {
    return this.value;
  }
}

File: transpile/javascript/util/DefaultHTMLImageProducer.js
/**
 * The default implementation of the AbstractHTMLImageProducer based on an
 * source (URL, base64, etc.)
 *
 * @author gianpiero.diblasi
 * @param <T> The value type
 */
class DefaultHTMLImageProducer extends AbstractHTMLImageProducer {

   src = null;

  /**
   * Creates the object
   *
   * @param value The value
   * @param src The source of the HTML image element (URL, base64, etc.)
   */
  constructor(value, src) {
    super(value);
    this.src = src;
  }

   produce() {
    let img = document.createElement("img");
    img.src = this.src;
    return img;
  }
}
